### PR TITLE
EDM-197: Set Condition on dev with multiple owners

### PR DESCRIPTION
**Pull request by avishayt**
Currently we use an annotation, which isn't optimal. Use a Condition instead.

### CODE REVIEW

File: internal/tasks/fleet_rollout.go [hunk 1]
```diff
@@ -123,11 +123,8 @@ func (f FleetRolloutsLogic) RolloutDevice(ctx context.Context) error {
 		return nil
 	}
 
-	if device.Metadata.Annotations != nil {
-		multipleOwners, ok := (*device.Metadata.Annotations)[model.DeviceAnnotationMultipleOwners]
-		if ok && len(multipleOwners) > 0 {
-			f.log.Warnf("Device has multiple owners, skipping rollout: %s", multipleOwners)
-		}
+	if api.IsStatusConditionTrue(device.Status.Conditions, api.DeviceMultipleOwners) {
+		f.log.Warnf("Device has multiple owners, skipping rollout")
```
Comment (hexfusion) [inline internal/tasks/fleet_rollout.go:127]: Wouldn't this be an error or can it be acceptable?

File: internal/tasks/fleet_selector.go [hunk 2]
```diff
@@ -450,19 +456,30 @@ func (f FleetSelectorMatchingLogic) findDeviceOwnerAmongAllFleets(ctx context.Co
...
 
 	err := f.setDeviceOwnerAccordingToMatchingFleets(ctx, device, currentOwnerFleet, matchingFleets)
 	if err != nil {
 		return nil, err
 	}
 
-	if currentAnnotationValue != newAnnotationValue {
-		annotations := map[string]string{
-			model.DeviceAnnotationMultipleOwners: newAnnotationValue,
+	f.log.Infof("HERE %s: %s - %s", *device.Metadata.Name, currentConditionMessage, newConditionMessage)
```
Comment (hexfusion) [inline internal/tasks/fleet_selector.go:471]: nit :)

File: internal/tasks/fleet_selector.go [hunk 3]
```diff
@@ -450,19 +456,30 @@ func (f FleetSelectorMatchingLogic) findDeviceOwnerAmongAllFleets(ctx context.Co
...
+	if currentConditionMessage != newConditionMessage {
+		condition := api.Condition{Type: api.DeviceMultipleOwners}
+		if len(matchingFleets) > 1 {
+			condition.Status = api.ConditionStatusTrue
+			condition.Reason = "MultipleOwners"
```
Comment (hexfusion) [inline internal/tasks/fleet_selector.go:476]: should the type be a higher level condition just feels extremely specific. For example. 

DeviceVerification
False
MultipleOwners
Msg

**Review by celdrake — COMMENTED** [review]
Tested in the UI, works well.
https://github.com/flightctl/flightctl-ui/pull/46

File: internal/tasks/fleet_selector.go (hunk 3 above)
Comment (avishayt) [reply internal/tasks/fleet_selector.go:476]: I see your point, but I'd like to make it clearer that the condition relates to the resources in the DB and not the actual device.  Also, "Device" in "DeviceVerification" feels redundant.

The device already has a "ValidSpec" condition.  If I generalize both of them, we want to know if the service is offering the latest spec to the agent.  If a device has multiple owners, the fleet logic won't even update the device's spec.  If the device.spec is invalid, the device logic won't offer the spec to the agent.  I can't think of a good name for this, but maybe that should be the scope of the condition.

Comment (avishayt) [reply internal/tasks/fleet_selector.go:476]: @hexfusion WDYT?

Comment (hexfusion) [reply internal/tasks/fleet_selector.go:476]: I am fine with it as is just something to think about.

**Review by hexfusion — APPROVED** [review]

**Review by celdrake — APPROVED** [review]

**Pull request merged**